appserver/transaction/jts/src/main/java/com/sun/jts/CosTransactions/RegisteredSyncs.java

```diff
@@ -113,6 +112,15 @@
      */
     RegisteredSyncs() {}
 
+    private RegisteredSyncs(List<Synchronization> rhs) {
+        registered.addAll(rhs);
+    }
+    
+    @Override
+    protected RegisteredSyncs clone() throws CloneNotSupportedException {
```

Comment:
If you'd dropped the throws clause here, you wouldn't need to handle it in TopCoordinator.